README.md

```diff
@@ -292,27 +292,31 @@ function UiComponent({ name, age }) {
     )
 }
 
-export default () => {
-    // you may extract these two lines into a custom hook
-    const store = useContext(StoreContext)
-    const data = userSelector(store)
-    return UiComponent({...data})
-    // perhaps wrap it inside observer in here?
-    return observer(UiComponent({...data}))
-}
+export default () => (
+    () => {
+        // you may extract these two lines into a custom hook
+        const store = useContext(StoreContext)
+        const data = userSelector(store)
+        return UiComponent({...data})
+        // perhaps wrap it inside observer in here?
+        return observer(UiComponent({...data}))
+    }
+)
 ```
 
-It may look a bit more verbose than a _classic_ inject, but there is nothing stopping you to make your own `inject` HOC which is so much easier since everything is just a funciton.
+It may look a bit more verbose than a _classic_ inject, but there is nothing stopping you to make your own `inject` HOC which is so much easier since everything is just a function.
 
 ```tsx
 // make universal HOC
 
-function inject(useSelector, baseComponent) {
-    const store = useContext(StoreContext)
-    const selected = useSelector(store)
-    // optional memo essentially making a pure component
-    return React.memo(props => baseComponent({ ...selected, ...props }))
-}
+const inject = (useSelector, baseComponent) => (
+    () => {
```

Comment:
I believe it should actually like this. With your approach, it's a component returning another component.

```suggestion
    React.useMemo((props) => {
        const store = useContext(StoreContext)
        const selected = useSelector(store)
        return baseComponent({ ...selected, ...props })
    })
```

Reply:
Thank you. Hopefully it's looking OK now :+1:

Reply:
Copy & paste again, you forgot one thing. And try to think about what it actually does :)

Reply:
Ops. Thanks for your patience. I'll be sure to do a lot more research!